src/components/SingleReviewInfo.jsx

```diff
@@ -1,25 +1,95 @@
 import { Link } from "react-router-dom";
+import { patchReviewUpVote, patchReviewDownVote } from "../utils";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
 
 export function SingleReviewInfo({
   category,
   created_at,
   designer,
-  owner,
   review_body,
-  review_id,
   review_img_url,
   title,
   votes,
+  setIndividualReview,
 }) {
+  const { review_id } = useParams();
+  const [err, setErr] = useState(false);
+
+  const upVote = (id) => {
+    setIndividualReview((currIndividualReview) => {
+      if (currIndividualReview.votes === votes) {
+        return { ...currIndividualReview, votes: votes + 1 };
+      }
+      return currIndividualReview;
+    });
+
+    patchReviewUpVote(id).catch((err) => {
+      setErr(true);
+      console.log(err.response.data);
+      setIndividualReview((currIndividualReview) => {
+        if (currIndividualReview.votes === votes) {
+          return { ...currIndividualReview, votes: votes - 1 };
+        }
+        return currIndividualReview;
+      });
+    });
+  };
+
+  const downVote = (id) => {
+    setIndividualReview((currIndividualReview) => {
+      if (currIndividualReview.votes === votes) {
+        return { ...currIndividualReview, votes: votes - 1 };
```

Comment:
At the moment I think we do not need this as in case of an error the votes are not visible anymore 🙂